### PR TITLE
[FIX] point_of_sale: better error when missing nomenclature

### PR DESCRIPTION
**Pull request by lse-odoo**
Before this commit:
If the PoS config have no "Barcode Nomenclature" set, the PoS session can't start and a cryptic JS traceback will be shown:
```js
TypeError: Cannot read properties of undefined (reading 'message')
at Chrome.start
```
This happen as the barcode rejected Promise request doesn't contains an error message. More details in:
https://github.com/odoo/odoo/pull/108797

After this commit:
A more user friendly error is displayed:
image

Note: the issue does not happen in previous Odoo's versions as the field was required before.

opw-3110204

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/js/Chrome.js [hunk 1]
```diff
@@ -201,6 +201,13 @@ odoo.define('point_of_sale.Chrome', function(require) {
         }
 
         setupBarcodeParser() {
+            if (!this.env.pos.company.nomenclature_id) {
+                const error = new Error("Missing barcode nomenclature");
+                console.error(error, error.stack);
+                error.stack = `The barcode nomenclature setting is not configured. \
+                Make sure to configure it on your Point of Sale configuration settings`;
+                throw error;
```
Comment (caburj) [inline addons/point_of_sale/static/src/js/Chrome.js:209]: I was just thinking of `throw new Error("Missing barcode nomenclature");`.

How about:

1. `throw new Error("Missing barcode nomenclature", { cause: { kind: "SETUP_BARCODE_ERROR", message: "<message here>" } })`.
2. in `Chrome.start`, explicitly handle this error. Show any `ErropPopup` you want and use the information from error.cause.

I know it's more code but you won't need to replace "stack" props in the error.

Comment (caburj) [reply addons/point_of_sale/static/src/js/Chrome.js:209]: And probably in the future, the "catch" clause in the `Chrome.start` can be standardize to handle different kinds of errors.

**Comment by lse-odoo** [comment]
@caburj 
Code updated! I didn't know about this trick for the Error, pretty handy!

I was thinking to use `ErrorPopup`, but, I think using a `ErrorTracebackPopup` is more relevant as the other exsting pop-up does not have `exitButtonIsShown` which can be handy + it will add complexity to the `catch` which is not necessary in my opinion. Let me know what you think of this version :)

File: addons/point_of_sale/static/src/js/Chrome.js [hunk 2]
```diff
@@ -201,6 +206,13 @@ odoo.define('point_of_sale.Chrome', function(require) {
         }
 
         setupBarcodeParser() {
+            if (!this.env.pos.company.nomenclature_id) {
+                const errorMessage = `The barcode nomenclature setting is not configured. \
+                Make sure to configure it on your Point of Sale configuration settings`;
+                throw new Error("Missing barcode nomenclature", { cause:
```
Comment (caburj) [inline addons/point_of_sale/static/src/js/Chrome.js:212]:
```suggestion
                throw new Error(this.env._t("Missing barcode nomenclature"), { cause:
```

**Review by caburj — CHANGES_REQUESTED** [review]
Nice. I have some comments/suggestions. After them we go.

**Comment by lse-odoo** [comment]
@caburj 
Let me know what you think of it now :)

File: addons/point_of_sale/static/src/js/Chrome.js [hunk 3]
```diff
@@ -201,6 +205,12 @@ odoo.define('point_of_sale.Chrome', function(require) {
         }
 
         setupBarcodeParser() {
+            if (!this.env.pos.company.nomenclature_id) {
+                const errorMessage = this.env._t("The barcode nomenclature setting is not configured. " +
+                    "Make sure to configure it on your Point of Sale configuration settings");
```
Comment (caburj) [inline addons/point_of_sale/static/src/js/Chrome.js:210]: Almost there. I know it doesn't look good, but AFAIK, only static string can be translated so it should be written as a single string.

```suggestion
                const errorMessage = this.env._t("The barcode nomenclature setting is not configured. Make sure to configure it on your Point of Sale configuration settings");
```

Comment (lse-odoo) [reply addons/point_of_sale/static/src/js/Chrome.js:210]: are you sure ? I saw some cases on multiples lines like:
https://github.com/odoo/odoo/blob/16.0/addons/point_of_sale/static/src/js/Misc/AbstractReceiptScreen.js#L50-L52

Comment (caburj) [reply addons/point_of_sale/static/src/js/Chrome.js:210]: True. I'm wrong. Looks like it's allowed: https://www.odoo.com/documentation/16.0/developer/howtos/translations.html#blocks

Comment (trj-odoo) [reply addons/point_of_sale/static/src/js/Chrome.js:210]: don't forget to include the pot file

**Review by caburj — APPROVED** [review]

**Comment by lse-odoo** [comment]
@caburj @trj-odoo should be good to go :)

File: addons/point_of_sale/i18n/point_of_sale.pot [hunk 4]
```diff
@@ -6,8 +6,8 @@ msgid ""
 msgstr ""
 "Project-Id-Version: Odoo Server 16.0+e\n"
 "Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2022-12-15 12:48+0000\n"
-"PO-Revision-Date: 2022-12-15 12:48+0000\n"
+"POT-Creation-Date: 2023-01-03 15:25+0000\n"
+"PO-Revision-Date: 2023-01-03 15:25+0000\n"
```
Comment (trj-odoo) [inline addons/point_of_sale/i18n/point_of_sale.pot:10]: can you remove this change ? I know it's automatically done but apparently it's a rule to omit this when modifying the pot file

**Pull request closed without merging**